### PR TITLE
EF-43: Support Enums

### PR DESCRIPTION
**Pull request by sanych-sun**

### CODE REVIEW

File: tests/MongoDB.EntityFrameworkCore.FunctionalTests/Mapping/ClrTypeMappingTests.cs [hunk 1]
```diff
@@ -434,12 +434,23 @@ public void Double_negative_read()
...
         var methodInfo = this.GetType().GetMethod(nameof(ClrTypeMappingTestImpl), BindingFlags.Instance | BindingFlags.NonPublic);
         methodInfo.MakeGenericMethod(valueType).Invoke(this, new[] { value });
     }
 
+    private enum TestEnum
+    {
+        EnumValue0 = 0,
+        EnumValue1 = 1
+    }
+
```
Comment (damieng) [inline tests/MongoDB.EntityFrameworkCore.FunctionalTests/Mapping/ClrTypeMappingTests.cs:453]: Can we include some test coverage for inserts of entities with Enums? 

Right now the test uses the Mongo C# Driver to insert the data for the test which is fine for exercising the query and materialization paths but insert/update path should also have coverage.

**Review by damieng — CHANGES_REQUESTED** [review]
We should probably also add support for configuring if the enum is stored in the database as a number or a string. (Also enums don't have to be just int's so we should consider test/support for those as well)

File: tests/MongoDB.EntityFrameworkCore.FunctionalTests/Mapping/ClrTypeMappingTests.cs (hunk 1 above)
Comment (sanych-sun) [reply tests/MongoDB.EntityFrameworkCore.FunctionalTests/Mapping/ClrTypeMappingTests.cs:458]: Done.

**Comment by sanych-sun** [comment]
More tests added. Regarding enum representation configuration - isn't it a part of the field tuning/configuration task? https://jira.mongodb.org/projects/EF/issues/EF-44

**Review by damieng — APPROVED** [review]

**Pull request merged**